File: shared/src/iosMain/kotlin/dev/chara/tasks/util/FriendlyDateFormat.kt
package dev.chara.tasks.util

import platform.Foundation.NSCalendar
import platform.Foundation.NSDate
import platform.Foundation.NSDateFormatter
import platform.Foundation.NSDateFormatterMediumStyle
import platform.Foundation.NSDateFormatterNoStyle
import platform.Foundation.NSDateFormatterShortStyle
import platform.Foundation.NSLocale
import platform.Foundation.currentLocale

class FriendlyDateFormat {
    fun formatDate(date: NSDate): String {
        val formatter = NSDateFormatter()

        formatter.dateStyle = NSDateFormatterMediumStyle
        formatter.timeStyle = NSDateFormatterNoStyle
        formatter.locale = NSLocale.currentLocale

        return formatter.stringFromDate(date)
    }

    fun formatDateTie(date: NSDate): String {
        val formatter = NSDateFormatter()

        formatter.dateStyle = if (NSCalendar.currentCalendar.isDateInToday(date)) {
            NSDateFormatterNoStyle
        } else {
            NSDateFormatterMediumStyle
        }
        formatter.timeStyle = NSDateFormatterShortStyle
        formatter.locale = NSLocale.currentLocale

        return formatter.stringFromDate(date)
    }
}

File: ios/Tasks/Util/FriendlyDateFormat.swift
//
//  FriendlyDateFormat.swift
//  Tasks
//
//  Created by Luke Myers on 4/12/23.
//  Copyright © 2023 orgName. All rights reserved.
//

import Foundation

class FriendlyDateFormat {
    func formatDate(date: Date) -> String {
        let formatter = DateFormatter()

        formatter.dateStyle = .medium
        formatter.timeStyle = .none
        formatter.locale = Locale.current

        return formatter.string(from: date)
    }

    func formatDateTime(date: Date) -> String {
        let formatter = DateFormatter()

        formatter.dateStyle = Calendar.current.isDateInToday(date) ? .none : .medium
        
        formatter.timeStyle = .short
        formatter.locale = Locale.current

        return formatter.string(from: date)
    }
}
